internal/setup/setup.go

```diff
@@ -157,12 +176,35 @@ func (r *Release) validate() error {
 						}
 						return fmt.Errorf("slices %s and %s conflict on %s", old, new, newPath)
 					}
-				} else {
-					if newInfo.Kind == GlobPath {
-						globs[newPath] = new
+					// Note: We do not have to record newPath because conflict
+					// is a transitive relation.
+					continue
+				}
+
+				// Check for glob and generate conflicts.
+				for oldPath, old := range paths {
+					oldInfo := old.Contents[oldPath]
+					if !(newInfo.Kind == GlobPath || newInfo.Kind == GeneratePath ||
+						oldInfo.Kind == GlobPath || oldInfo.Kind == GeneratePath) {
+						continue
```

Comment:
The overall loop here was de-optimized, probably in an unnecessary way.

The for loops above are already executing: for every file (content), inside every slice, of every package. So already quite a relevant expansion.  Now the new code is also adding almost every one of those items to a list, and for every one inner iteration of the three earlier loops, it's looping over that whole list again. As a quick exercise, assume 10k items in the earlier loops, how many times are we executing the logic here?  How many times did we go through the exact same items before rejecting them? (hint: just for the first element of the list, 10k-1 times).

That's why the original code had a `globs` helper here. The cost was similar, but we were paying only for items that we knew had to be handled as globs. I think we still want something similar, but need different conditions for its use as you've spotted.

Reply:
Discussed offline, even if GlobPath is the most expensive operation here the approach in the PR does not make sense. The code that was in the PR was going to do roughly `O(n^2)` loop iterations (combinatorial explosion) while the previous code was using `globs` to reduce that by the % of globs in the slice definitions. For example, given 24.04 from chisel-releases that has ~25% globs (as of today), the previous algorithm takes, in theory, 1/4 of the time of the new one. This is especially relevant for use-cases where the % is even lower, which is what we envision for the future of the releases. The only cost is one extra map which is pretty reasonable.

I have changed the code to "tweak" the previous algorithm while solving the bugs and adding the support for generate. I am only not sure about the naming of the new map, but that is a very minor thing.